CMakeLists.txt

```diff
@@ -57,6 +61,10 @@ endif()
 # Older stdc++ enable c++11 items
 add_definitions(-D__USE_XOPEN2K8)
 
+check_symbol_exists(fopen64 stdio.h FOPEN_64_PRESENT)
+message(STATUS "${CMAKE_LOCAL}/build_config.h.in -> ${INCLUDE_DMLC_DIR}/build_config.h")
```

Comment:
Why?

Reply:
@larroy Some systems like MinGW have fopen() and fopen64(), and we want to choose fopen64() to use 64-bit offsets

Reply:
Thanks for your response, I didn't think you would read it. The problem with this change is that build_config.h is recreated on build, which is marking the 3rdparty submodule dirty in mxnet everytime we build. Is this build_config.h changed by CMake neccessary? I would like to avoid making the repository dirty on build.

Thanks.

Reply:
@larroy Originally, I had planned to have CMake create build_config.h on the fly. The header file contains special macros that are specific to platforms, e.g. existence of fopen64(). The issue was that not everyone uses CMake; many projects using dmlc-core uses GNU Make instead. So I created build_config.h containing the "default logic". Users of GNU Make will use the header as it is. On the other hand, CMake users will overwrite the header, to reflect the platform-specific details that CMake detects.

Reply:
@larroy One solution would be to add build_config.h to .gitignore.

Reply:
@hcho3 Thanks for the explanation. Another option is to generate the file with both systems as a configure step. That last one seems more clean to me, so do I understand correctly that the header is there under version control because of Make builds?

Reply:
Isn't it already in ignore?  https://github.com/dmlc/dmlc-core/blob/master/.gitignore#L49

Reply:
@larroy Yes, that's correct. Keep in mind that not every user of GNU Make uses Autotools ("configure")

Reply:
@larroy That's weird, maybe path is off?

Reply:
I think it won't work if the file is in the repo:

I tried locally.

https://stackoverflow.com/questions/1818895/keep-ignored-files-out-of-git-status

Reply:
Got it. I'm open to suggestion as to how to better accommodate both CMake and GNU Make.

Reply:
For now, you should run
```
git update-index --assume-unchanged include/dmlc/build_config.h
```
from [this page](https://wildlyinaccurate.com/git-ignore-changes-in-already-tracked-files/)